frontend/tet/admin/src/components/BackButton.tsx

```diff
@@ -21,10 +23,12 @@ const BackButton: React.FC = () => {
   };
 
   return (
-    <$BackButton onClick={clickHandler}>
-      <IconArrowLeft />
-      <span>{t('common:application.backToFrontPage')}</span>
-    </$BackButton>
+    <Link href="/" passHref>
```

Comment:
What about using [LinkButton](https://github.com/City-of-Helsinki/yjdh/tree/develop/frontend/shared/src/components/link-button) ? It looks like a link but it's a button in accessibility-wise. It also supports the `iconLeft` property.